lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -155,223 +182,188 @@
+        if (extensionFilter !== undefined) {
+            var filtered = [];
+            for (let i = 0; i < subTags.length; i++) {
+                const tag = subTags[i];
+                if (_.indexOf(extensionFilter, tag) !== -1) {
+                    // tag was an expected unicode extension key, check the value
+                    const tagValue = subTags[i + 1];
+                    if (i + 1 < subTags.length && tagValue.length > 2) {
+                        supportedExtensionAddition += `-${tag}-${tagValue}`;
+                        i += 1;
+                    }
                 }
             }
 
-            resolved += callInstanceFunc(ArrayInstanceJoin, subTags, "-");
-        } else {
-            resolved = undefined;
+            supportedExtension += supportedExtensionAddition;
         }
 
-        return setPrototype({
-            locale: resolved,
+        let resolved = resolvedWithoutSubtags;
+
+        return bare({
+            locale: supportedExtension.length > 2 ? resolvedWithoutSubtags + supportedExtension : resolvedWithoutSubtags,
             subTags: subTags,
-            localeWithoutSubtags: withoutSubTags
-        }, null);
+            localeWithoutSubtags: resolvedWithoutSubtags
+        });
     }
 
-    var resolveLocales = function (givenLocales, matcher, extensionFilter, defaultLocaleFunc) {
-        var fitter = matcher === "lookup" ? resolveLocaleLookup : resolveLocaleBestFit;
-        var length = getArrayLength(givenLocales);
+    var resolveLocales = function (givenLocales, matcher, extensionFilter) {
+        const fitter = matcher === "lookup" ? resolveLocaleLookup : resolveLocaleBestFit;
+        const length = getArrayLength(givenLocales) || 0;
 
-        var defaultLocale = defaultLocaleFunc();
-
-        length = length !== undefined ? length : 0;
-        for (var i = 0; i < length; i++) {
-            var resolved = resolveLocaleHelper(givenLocales[i], fitter, extensionFilter, defaultLocale);
-            if (resolved.locale !== undefined) {
+        const defaultLocale = GetDefaultLocale();
+        for (let i = 0; i < length; i++) {
+            const resolved = resolveLocaleHelper(givenLocales[i], fitter, extensionFilter);
+            if (resolved && resolved.locale !== undefined) {
                 return resolved;
             }
         }
-        return resolveLocaleHelper(defaultLocale, fitter, undefined, defaultLocale);
-    }
-
-    // get just the language-script-region from the default locale
-    let __strippedDefaultLocale = undefined;
-    var strippedDefaultLocale = function () {
-        if (__strippedDefaultLocale) {
-            return __strippedDefaultLocale;
-        }
+        return resolveLocaleHelper(defaultLocale, fitter, extensionFilter);
+    };
 
-        if (isPlatformUsingICU) {
-            if (!LANG_TAG_BASE_RE) {
-                InitializeLangTagREs();
+    var Internal = bare({
+        ToObject: function (o) {
```

Comment:
These properties can also be members:

```js
var Internal = bare({
    ToObject(o) {
        if (o === null) {
        ...
```

Reply:
Yeah, I definitely noticed these as candidates for that change, but there was so much altered here that I didn't want to make the diff any more noisy than it already was (while I am fully aware that I didn't always follow that rule, that what I was thinking here). Honestly a lot of these helpers could probably be refactored -- what is "Internal", why are the method names uppercase rather than lowercase, etc (I have only been using pascal-case for spec-defined algorithms while all other helpers get camelcase)

Reply:
Feel free to rename these according to camelCase rules. I'll sign off on this. 👍